managers/asset-operations-manager.js

```diff
@@ -345,43 +345,57 @@ class AssetOperationsManager {
 
         const { tokenId } = resolveUAL(UAL);
 
-        let publicAssertionId;
-        if(state === ASSET_STATES.LATEST) {
+        let publicAssertionId,
+            stateFinalized = false;
+        if (state === ASSET_STATES.LATEST) {
             const unfinalizedState = await this.blockchainService.getUnfinalizedState(
                 tokenId,
                 blockchain,
             );
 
             if (unfinalizedState != null && unfinalizedState !== ZeroHash) {
                 publicAssertionId = unfinalizedState;
+                stateFinalized = false;
+            } else {
+                stateFinalized = true;
             }
         }
 
         let assertionIds = [];
         const isEnumState = Object.values(ASSET_STATES).includes(state);
-        if(!publicAssertionId) {
+        if (!publicAssertionId) {
             assertionIds = await this.blockchainService.getAssertionIds(tokenId, blockchain);
 
             if (isEnumState) {
                 publicAssertionId = assertionIds[assertionIds.length - 1];
+                stateFinalized = true;
```

Comment:
Here basically we already know that this state is finalized even if we chose the LATEST options, all good